tests/system/test_dbapi.py

```diff
@@ -536,6 +536,90 @@ def test_batch_dml_invalid_statements(self):
         with pytest.raises(OperationalError):
             self._cursor.execute("run batch")
 
+    def test_partitioned_query(self):
+        """Test partition query works in read-only mode."""
+        self._cursor.execute("start batch dml")
+        for i in range(1, 11):
+            self._insert_row(i)
+        self._cursor.execute("run batch")
+        self._conn.commit()
+
+        self._conn.read_only = True
+        self._cursor.execute("PARTITION SELECT * FROM contacts")
+        partition_id_rows = self._cursor.fetchall()
+        assert len(partition_id_rows) > 0
+
+        rows = []
+        for partition_id_row in partition_id_rows:
+            self._cursor.execute("RUN PARTITION " + partition_id_row[0])
+            rows = rows + self._cursor.fetchall()
+        assert len(rows) == 10
+        self._conn.commit()
+
+    def test_partitioned_query_in_rw_transaction(self):
+        """Test partition query throws exception when connection is not in
+        read-only mode and neither in auto-commit mode."""
+        self._cursor.execute("start batch dml")
+        for i in range(1, 11):
+            self._insert_row(i)
+        self._cursor.execute("run batch")
+        self._conn.commit()
```

Comment:
nit: You can skip this part in this test, as it is not needed

Reply:
Done